cep-techspec.md

```diff
@@ -0,0 +1,339 @@
+<table>
+<tr><td> Title </td><td> Technical specification for creation and deletion of conda environments </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jrodriguez@quansight.com&gt;</td></tr>
+<tr><td> Created </td><td> Jun 10, 2022</td></tr>
+<tr><td> Updated </td><td> Jun 10, 2022</td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+Conda environments have existed for more than 10 years. During most of that time,
+`conda` was the only tool available to create, modify and delete conda environments. As a result,
+once the Python implementation was considered stable enough, its integration test suite could be
+regarded as the only technical specification needed for practical purposes.
+
+However, as other tools appeared in the ecosystem (namely `mamba` and `micromamba`, along with the
+`libmamba`-based solver), some of the tests contained in the integration suite did not adjust to
+the ecosystem expectations (tests too strict, too tied to `conda` implementation details,
+ambiguous results, conflicting outcomes across different tests, etc).
+
+In this CEP, we'd like to propose a minimum set of guidelines any software implementation that
+operates on `conda` environments should adhere to.
+
+## Specification
+
+The technical specification will list a set of cases where known input conditions are given, and
+list the expected outcome and its rationale. It will be detailed at the command-line interface
+level following conda's 4.13 implementation. However, this CEP will not try to standardize the CLI
+commands, subcommands or flags. That work can be discussed in a different CEP.
+
+### Terminology
+
+#### Conda environment
+
+The [conda docs][conda-docs-environments] define a `conda` environment as a "a directory that
+contains a specific collection of conda packages that you have installed". However, the internal
+structure of that directory is not fully specified, since it depends on the contents of the
+packages installed.
+
+The only technical requirement enforced by `conda` activation mechanisms is that the environment
+must contain a `conda-meta` directory with a `history` file. Other than that, the environment can
+contain anything. It must be noted, though, that there are other important locations which might be
+present in the directory tree. These are optional, but have a role in the conda environment
+lifecycle:
+
+- `bin/`: added to PATH on Unix
+- `lib/`: searched for dynamic libraries on Unix
```

Comment:
@jaimergp we try our best to force packages to use `lib` but sometimes `lib64` passes without our knowledge. Maybe, if we write this down in the spec, we should start warning/breaking packages that deviates from this at build time.

Reply:
Agreed! I've been thinking about resurrecting `conda-verify` (maybe a subset of the checks), and including some of the stuff mentioned in https://github.com/conda-forge/conda-forge.github.io/pull/1784 to make it easier for the reviewer.